### ISSUE TITLE
[entry] in flow and twin 

### ISSUE THREAD

**Issue by rebekkategtmeier** [andriisemenov/equihub-brain-issue-tracking#104]
Me as a user, 
would like to transform a small and fast entry into a story (activity) so that I can work on the realization or the validation of it. 

Outcomes: 

Twin: 
- [ ] I can transform a small entry into a "story" which I can put in my sprint planning
- [ ] the entry is automatically the outcome of the story 
- [ ] show validation criteria in twin after the transformation of the story to input the findings (side bar)
- [ ] I can enter the validation status and findings, when the story is done 
- [ ] A validation cannot be saved without the finding

In the flow
- [ ]  FAB with label: create entry 
- [ ] opens the new and empty story template which is related to the twin (e.g. choose persona / outcome from twin ) or generate a completely new entry 
- [ ] new outcome of the new entry will be shown in twin 
- [ ] story can be saved without defining an outcome for the twin (important for the education part)
- [ ] the additional entry can be put to a sprint

**Comment by rajashreekendre** [comment]
> Rejashree:
> 
> Editing hypothesesis: id is required, all other things optional.

Done with the changes, id is mandatory and all other fields are optional for edit hypothesis

**Comment by andriisemenov** [comment]
@rajashreekendre 
Add a field in hypothesis -> outcomes (one-to-many hypotheses)
Method -> adding a hypothesis to hypothesis as an outcome

**Comment by rebekkategtmeier** [comment]
Decision was to start with the integration of the outcomes first and than decide how to go on next.

**Comment by rajashreekendre** [comment]
> @rajashreekendre Add a field in hypothesis -> outcomes (one-to-many hypotheses) Method -> adding a hypothesis to hypothesis as an outcome

**Comment by rajashreekendre** [comment]
> @rajashreekendre Add a field in hypothesis -> outcomes (one-to-many hypotheses) Method -> adding a hypothesis to hypothesis as an outcome

**Comment by rajashreekendre** [comment]
> @rajashreekendre Add a field in hypothesis -> outcomes (many-to-many hypothesis) Method -> adding a hypothesis to hypothesis as an outcome

![image](https://user-images.githubusercontent.com/29223247/141536436-500c33e0-120a-42e6-a42e-fbffbb7a9244.png)

**Comment by kochjh** [comment]
I need a possibility to get all Outcomes from a hypothesis. I think this functionality is missing, isn't it?

Adding outcomes to a hypothesis works. Great job!

**Comment by andriisemenov** [comment]
@rajashreekendre : Add fetching the outcomes from a hypothesis by hypothesis id.

**Comment by rajashreekendre** [comment]
> @rajashreekendre :
> 
> * Add fetching the outcomes from a hypothesis by hypothesis id (only hypothesis id is needed).
> * Removing the outcome connection to the hypothesis. removeHypothesisAsOutcomes(id:"", outcomes:[])

To check this functionality please use below graphql operations:
1. Fetch the outcome:
  query { getListOfOutcomes(id:"feed7d69-25c5-4684-99e4-dfb687d3404f"){
  id,
  title
}}

2. To remove one/multiple outcomes : 
mutation{ removeHypothesisAsOutcomes(id:"feed7d69-25c5-4684-99e4-dfb687d3404f",
outcomes:["e348f60f-a8e1-4d6a-b282-608c1da999b1","64c7bec9-0fa8-4d3f-b01f-7027380183eb","6d3700c7-f51d-4c4f-a1a5-483072c2d519"]){
id
}
}

Note: outcomes are the list of hypothesis Ids

**Comment by kochjh** [comment]
Jan & Rebekka Meeting, results and open todos:
- Entry : Story is a 1:n relation. (Story is created by max. 1 entry, an entry can create multiple stories)
- entries should display the stories they created, there is also the possibility to create a new story from this entry
- The new story functionality shouldn't be in a new dialog, it should "slide into" the same dialog (editHypothesis).

 @andriisemenov @rajashreekendre I need a query, which returns all stories given a hypothesis (So all stories which are linked to these hypothesis). Can we have a short meeting to discuss this?

**Comment by andriisemenov** [comment]
@kochjh : I see what you mean now. Yes, we need to create a new function in the backend to give the substeps of a given hypothesis.
@rajashreekendre : Please, create a function taking an id of the hypothesis and outputting an array of substeps connected to this hypothesis.

**Comment by rebekkategtmeier** [comment]
blocked by two backend things which will be done later. 
Finish this week to review it on Wednesday. 

Could be tested with the students on the 14th.

**Comment by kochjh** [comment]
Backend:
- [ ] Resolve bug in adding outcomes

**Comment by kochjh** [comment]
- [ ] when clicking enter without input the 3-month things shoudl not be shown 
- [ ] margin in the end when the story is put in 
- [ ] enter after the outcomes instead of the list icon

**Comment by rebekkategtmeier** [comment]
@rajashreekendre  and @kochjh we should unlink the story when the entry will be deleted

**Comment by rajashreekendre** [comment]
> @rajashreekendre and @kochjh we should unlink the story when the entry will be deleted

 Now, we unlinking the story when the entry is deleted

**Comment by kochjh** [comment]
@andriisemenov we can merge it
